corehq/motech/generic_inbound/utils.py

```diff
@@ -19,18 +20,31 @@ def get_context_from_request(request):
     except (UnicodeDecodeError, json.JSONDecodeError):
         raise GenericInboundUserError(_("Payload must be valid JSON"))
 
+    couch_user = request.couch_user
+    if couch_user.is_commcare_user():
+        restore_user = couch_user.to_ota_restore_user(couch_user)
+    elif couch_user.is_web_user():
+        restore_user = couch_user.to_ota_restore_user(request.domain, couch_user)
+    else:
+        raise GenericInboundUserError(_("Unknown user type"))
```

Comment:
this logic will go away once https://github.com/dimagi/commcare-hq/pull/31876/commits/39dfc4566277027ba0e38e96cc6d15f08d035427#diff-72cb8ba3bb5b51c7450986f76285fc191501fc01ba292789849e90d794fb7662R1737 is merged